core/java/com/google/instrumentation/stats/DistributionAggregation.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2016, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.instrumentation.stats;
+
+import com.google.instrumentation.common.Duration;
+import com.google.instrumentation.common.Function;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.Nullable;
+
+/**
+ * An aggregation of data based on distributions.
+ *
+ * <p> A distribution contains summary statistics for a population of values and, optionally, a
+ * histogram representing the distribution of those values across a specified set of histogram
+ * buckets, as defined in {@link DistributionAggregationDescriptor#getBucketBoundaries()}.
+ *
+ * <p>Although not forbidden, it is generally a bad idea to include non-finite values (infinities
+ * or NaNs) in the population of values, as this will render the {@code mean} meaningless.
+ */
+public final class DistributionAggregation {
+  /**
+   * Constructs a new {@link DistributionAggregation}.
+   */
+  public static final DistributionAggregation create(
```

Comment:
I'm on the border here with changing this to a builder but I'm going to keep it a static factory for now to stay analogous with IntervalAggregation. I decided to move tags to the end, just before the option bucket counts.